### ISSUE TITLE
[CoE Starter Kit - BUG] ISSUE TITLE - Dataflows are not getting refreshed automatically after the Sep 2024 Upgrade for Test Governance Env.

### ISSUE THREAD

**Issue by Kumarez007** [microsoft/coe-starter-kit#8987]
### Does this bug already exist in our backlog?

- [X] I have checked and confirm this is a new bug.

### Describe the issue

Dataflows are not getting refreshed automatically after the Sep 2024 Upgrade for Test Governance Env. I tried to check the cloudflows and the status is in Off. When I try to turn it On it gives me the below error.

Turn on failed. Flow client error returned with status code "BadRequest" and details "{"error":{"code":"XrmEnvironmentVariableAttributeNotFound","message":"Attribute 'value' was not found for environment variable 'admin_ModelAppDataflowID'."}}".
Turn on failed. Flow client error returned with status code "BadRequest" and details "{"error":{"code":"XrmEnvironmentVariableAttributeNotFound","message":"Attribute 'value' was not found for environment variable 'admin_AppDataflowID'."}}".
Turn on failed. Flow client error returned with status code "BadRequest" and details "{"error":{"code":"XrmEnvironmentVariableAttributeNotFound","message":"Attribute 'value' was not found for environment variable 'admin_FlowDataflowID'."}}".
Turn on failed. Flow client error returned with status code "BadRequest" and details "{"error":{"code":"XrmEnvironmentVariableAttributeNotFound","message":"Attribute 'value' was not found for environment variable 'admin_FlowDataflowID'."}}".
 Turn on failed. Flow client error returned with status code "BadRequest" and details "{"error":{"code":"XrmEnvironmentVariableAttributeNotFound","message":"Attribute 'value' was not found for environment variable 'admin_EnvironmentDataflowID'."}}".

### What solution are you experiencing the issue with?

Governance

### What solution version are you using?

4.43

### What app or flow are you having the issue with?

Dataflows

### What method are you using to get inventory and telemetry?

Cloud flows

**Comment by Kumarez007** [comment]
![Issue](https://github.com/user-attachments/assets/96e4beb9-5235-4f84-aac0-e34d4d1c0b3a)

**Comment by Kumarez007** [comment]
Can I get an update on this please

**Comment by Jenefer-Monroe** [comment]
These env variables are supposed to be filled by the setup wizard. Did you run that?

**Comment by Kumarez007** [comment]
Hi Jenefer, Yes I tried there as well but it didn't turn on when I did the upgrade

**Comment by Kumarez007** [comment]
Is there a solution to turn it on again now?

**Comment by Jenefer-Monroe** [comment]
Do you have values for all these env vars? 
![image](https://github.com/user-attachments/assets/5a0e001b-5485-4ff7-8449-eff7a9ce49f7)

As well as this one:
![image](https://github.com/user-attachments/assets/1c7b6960-54bc-419d-8bda-890eba62de3b)

**Comment by Kumarez007** [comment]
the dataflow value in use is empty in our environment. Please let me know the next step for the solution.

**Comment by Jenefer-Monroe** [comment]
In core you will see a flow called **_DebugBYODL>ConfigureEnvironment_**

Open it for edit and chose your envt and your Maker dataflow, save, and turn on.
![image](https://github.com/user-attachments/assets/4fd9fd20-a31e-44c9-a1fc-7fd8f28b48c6)

Go manually trigger the Maker Dataflow
![image](https://github.com/user-attachments/assets/75a7d0dc-a23e-4558-bfff-8238aa23cbf0)

When the dataflow completes this flow will trigger, open the run and you will find the Dataflow ID for Maker in the trigger output
![image](https://github.com/user-attachments/assets/626ef644-40f2-4b1a-aea3-3484ea9a952f)

While here also validate that the Current Environment one is correct:
![image](https://github.com/user-attachments/assets/a702c29f-a92e-4983-bcb6-75cc40107488)

Now do the same for the rest of the dataflows in the following order
- Environments
- Apps
- Model Driven Apps
- Flows
- App Connections
- Apps Last Launched Date
- Flow Connections
- Flow Last Run Date 

When you have all the environment variables filled, be sure to turn back off **_DebugBYODL>ConfigureEnvironment_** to avoid it needlessly running. 
And now you can turn on those flows.

**Comment by Kumarez007** [comment]
I have followed the above mentioned steps and the Env variables are still not added and I am not able to turn on the dataflows for automatic refresh. please let me know if anything else we can do.

**Comment by Jenefer-Monroe** [comment]
you must enter the env vars yourself. That was the way to get the values to enter.

**Comment by Jenefer-Monroe** [comment]
closing out as no further action for starter kit team

**Comment by Kumarez007** [comment]
Hi Jenefer,

Even after adding the value it didn't resolve the issue and also the values gets disappeared again after it was entered manually. Any other way to resolve this issue?

**Comment by Jenefer-Monroe** [comment]
Please share a screenshot of how you added the env var value. What UX did you use?

**Comment by Kumarez007** [comment]
ok thanks... let me share you the screenshot now

**Comment by Kumarez007** [comment]
![image](https://github.com/user-attachments/assets/37fc0135-6bb9-4174-b8e9-f0372169db0b)

This is before adding the variables

**Comment by Jenefer-Monroe** [comment]
Did you use this app to update the env vars? 
I noticed you are in edit mode, why? I wonder if you have an unmanaged layer on this page as a result of being in edit mode.

**Comment by Kumarez007** [comment]
This is after adding the variables
![image](https://github.com/user-attachments/assets/b511100b-11a1-453e-8301-6aabe18a5fba)

**Comment by Kumarez007** [comment]
> Did you use this app to update the env vars? I noticed you are in edit mode, why? I wonder if you have an unmanaged layer on this page as a result of being in edit mode.

I have opened it to add the variables just now and took the screenshot

**Comment by Kumarez007** [comment]
Now I Have started the makers dataflow to refresh manually..

![image](https://github.com/user-attachments/assets/956bf30b-dd47-4b73-ab8f-5532d0918086)

**Comment by Kumarez007** [comment]
and checked the env variables.. It has vanished. please see the screenshot.

![image](https://github.com/user-attachments/assets/ec1ef3c8-a548-45c5-821a-ca4cc9f6031b)

**Comment by Kumarez007** [comment]
Hi Jenefer,

I tried adding the env variable againa nd went to bookmark page and came back to env varibales.. it is again vanished.

![image](https://github.com/user-attachments/assets/5a32087d-bb5e-4749-8be6-8da94e85980a)

**Comment by Kumarez007** [comment]
Will we be able to resolve this during this month Upgrade?

**Comment by Jenefer-Monroe** [comment]
Yep so if you are Sept release then you must have unmanaged layers present on that page. There was a product change in how to set env vars from canvas and we fixed it for Sept. 
I suspect you have them as you seem to open the app in edit mode. 
Please check for unmanaged layers on that page and remove them and then try again. 
![image](https://github.com/user-attachments/assets/d3b98ce9-4d0d-47e8-ab7c-2cc281b2ab99)

If that does not work you can just set them in the product UX itself. Be sure to set Current value not Default. 
![image](https://github.com/user-attachments/assets/460f47a1-46cf-4396-aa3f-66c9b50bad2f)

**Comment by Kumarez007** [comment]
I don't see any unmanaged layers

![image](https://github.com/user-attachments/assets/3f728108-a3dc-439d-ba33-779ea1e1a0e9)

**Comment by Jenefer-Monroe** [comment]
ok please just go ahead and update the env vars in the product UX itself to unblock yourself.

**Comment by Kumarez007** [comment]
Yes did it and it worked and I am able to turn on the flows now as below.

![image](https://github.com/user-attachments/assets/5cc3628e-6c4c-4a65-a433-08e0aaa7deec)

**Comment by AmarSaiRam** [comment]
Dear User, 

We noticed that this issue has been pending for quite some time, and we have not received any recent updates from your end. In the absence of further communication, we are proceeding to close this ticket, assuming that your concerns have been addressed or resolved. 
If you continue to experience any difficulties related to this matter, please feel free to open a new support ticket. This will help us ensure prompt and effective assistance, rather than reopening the new Issue. 
Thank you for your understanding and cooperation. 

Best Regards, 
CoE Starter Kit Team

**Comment by AmarSaiRam** [comment]
We haven’t heard back in a while, so we’ll proceed to close this issue for now. If the issue persists or you need further assistance, feel free to log a new issue. We’re happy to help anytime. Thanks again for your engagement!
